src/Domain/Model/Task/Task.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Domain\Model\Task;
+
+use Assert\Assert;
+
+final readonly class Task
+{
+    public function __construct(
+        private int $id,
+        private string $title,
+        private bool $completed,
+    ) {
+        Assert::that($title)->notEmpty("Title can't be an empty string");
+    }
+
+    public function id(): int
+    {
+        return $this->id;
+    }
+
+    public function title(): string
+    {
+        return $this->title;
+    }
+
+    public function isCompleted(): bool
+    {
+        return $this->completed;
+    }
```

Comment:
As long as the class is readonly, you can safely expose the properties publicly. However, at some point you need to think about how you can mutate the state of `completed` and `title`. Entities are mutable, and you need to be able to mark a task as completed. The `id` property will never change, so that's a very likely candidate to remain readonly.

It's a great example on how you manage state changes in an entity. I'm looking forward to see how you've tackled this challenge 😉

Reply:
I postponed this a bit, but yes, you're right!
I chose to separate a write and read model, and a way to retrieve either of them using the same repository.
Curious to what you think!